deps of deps not updating

#447 was failing because beats' dependency golang.org/x/crypto is being updated, golang.org/x/crypto/'s dependency on golang.org/x/sys/unix is not.

steps to reproduce:

```
# in apm-server
## fda7d0a15 is current apm-server master HEAD
## e19f2a6f6 is current beats master HEAD
git checkout fda7d0a15fea6d86e33a55dbc3591e4956925451:
govendor fetch github.com/elastic/beats/...@e19f2a6f639496d94eaee8cb06ec2c600766563f
```

updates:
```
vendor/github.com/elastic/beats/...
vendor/go.uber.org/
vendor/golang.org/x/crypto/
vendor/golang.org/x/sys/windows/registry/
vendor/golang.org/x/sys/windows/svc/eventlog/
```

and fails:
```
$ go test .
# github.com/elastic/apm-server/vendor/golang.org/x/crypto/ssh/terminal
vendor/golang.org/x/crypto/ssh/terminal/util.go:30:12: undefined: unix.IoctlGetTermios
vendor/golang.org/x/crypto/ssh/terminal/util.go:38:18: undefined: unix.IoctlGetTermios
vendor/golang.org/x/crypto/ssh/terminal/util.go:54:12: undefined: unix.IoctlSetTermios
vendor/golang.org/x/crypto/ssh/terminal/util.go:64:18: undefined: unix.IoctlGetTermios
vendor/golang.org/x/crypto/ssh/terminal/util.go:75:9: undefined: unix.IoctlSetTermios
vendor/golang.org/x/crypto/ssh/terminal/util.go:80:13: undefined: unix.IoctlGetWinsize
vendor/golang.org/x/crypto/ssh/terminal/util.go:98:18: undefined: unix.IoctlGetTermios
vendor/golang.org/x/crypto/ssh/terminal/util.go:107:12: undefined: unix.IoctlSetTermios
vendor/golang.org/x/crypto/ssh/terminal/util.go:112:3: undefined: unix.IoctlSetTermios
FAIL	github.com/elastic/apm-server [build failed]
```

those were introduced in a newer golang.org/x/sys than apm has vendored.  Updating that using the same **beats** commit fixes it:

```
## e19f2a6f6 is current beats master HEAD
govendor fetch golang.org/x/sys/unix@e19f2a6f639496d94eaee8cb06ec2c600766563f
$ go test .
ok  	github.com/elastic/apm-server	0.038s
```

--- a/script/update_beats.sh
+++ b/script/update_beats.sh
@@ -1,17 +1,24 @@
-#!/usr/bin/env bash -ex
+#!/usr/bin/env bash
+
+set -ex
 
 BEATS_VERSION="${BEATS_VERSION:-master}"
 
 # Find basedir and change to it
-BASEDIR=$(dirname "$0")/../_beats
+DIRNAME=$(dirname "$0")
+BASEDIR=${DIRNAME}/../_beats
 mkdir -p $BASEDIR
-cd $BASEDIR
+pushd $BASEDIR
 
 # Check out beats repo for updating
 GIT_CLONE=repo
-trap "{ rm -rf ${GIT_CLONE}; }" EXIT
+trap "{ rm -rf ${BASEDIR}/${GIT_CLONE}; }" EXIT
 
-git clone --depth 1 --branch ${BEATS_VERSION} https://github.com/elastic/beats.git ${GIT_CLONE}
+git clone https://github.com/elastic/beats.git ${GIT_CLONE}
+(
+    cd ${GIT_CLONE}
+    git checkout ${BEATS_VERSION}
+)
 
 # sync
 rsync -crpv --delete \
@@ -30,3 +37,9 @@ rsync -crpv --delete \
     --include="testing/***" \
     --exclude="*" \
     ${GIT_CLONE}/ .
+
+popd
+
+# use exactly the same beats revision rather than $BEATS_VERSION
+BEATS_REVISION=$(GIT_DIR=${BASEDIR}/${GIT_CLONE}/.git git rev-parse HEAD)
+${DIRNAME}/update_govendor_deps.py ${BEATS_REVISION}

--- a/script/update_beats.sh
+++ b/script/update_beats.sh
@@ -12,7 +12,7 @@ pushd $BASEDIR
 
 # Check out beats repo for updating
 GIT_CLONE=repo
-trap "{ rm -rf ${BASEDIR}/${GIT_CLONE}; }" EXIT
+trap "{ set +e;popd >/dev/null;set -e;rm -rf ${BASEDIR}/${GIT_CLONE}; }" EXIT
 
 git clone https://github.com/elastic/beats.git ${GIT_CLONE}
 (

--- a/script/update_govendor_deps.py
+++ b/script/update_govendor_deps.py
@@ -1,0 +1,52 @@
+#!/usr/bin/env python
+"""
+Update dependencies of the given govendor'd package.  For https://github.com/elastic/apm-server/issues/450.
+
+"""
+from __future__ import print_function
+
+import argparse
+import json
+import os
+import subprocess
+import sys
+
+
+def find_packages(fp, origin, revision):
+    """generate list of packages for the given origin prefix not already on revision"""
+    j = json.load(fp)
+    for pkg in j['package']:
+        if pkg['revision'] != revision and pkg.get('origin', '').startswith(origin):
+            yield pkg['path']
+
+
+def main():
+    default_vendor_file = os.path.join(os.path.abspath(os.path.dirname(sys.argv[0])), '..', 'vendor', 'vendor.json')
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-f', '--vendor-file',
+                        default=default_vendor_file, type=argparse.FileType(mode='r'), help='path to vendor.json')
+    parser.add_argument('-n', '--dryrun', action='store_true',
+                        help='do not perform updates, just list what would be updated')
+    parser.add_argument('--origin', default='github.com/elastic/beats/', help='update deps of this origin prefix')
+    parser.add_argument('-q', '--quiet', dest='verbose', action='store_false', help='work quietly')
+    parser.add_argument('revision', help='update deps to this revision')
+    args = parser.parse_args()
+
+    packages = find_packages(args.vendor_file, args.origin, args.revision)
+
+    if args.dryrun:
+        for pkg in sorted(packages):
+            print(pkg)
+        return
+
+    # much faster to call govendor once and minimize the git reset & status calls it makes
+    update = ['{pkg}@{revision}'.format(pkg=pkg, revision=args.revision) for pkg in packages]
+    cmd = ['govendor', 'fetch'] + update
+    if args.verbose:
+        print(' '.join(cmd))
+    subprocess.check_call(cmd)
+
+
+if __name__ == '__main__':
+    main()

